adapter/src/main/java/ar/edu/undec/adapter/service/curso/config/CursoBeanConfig.java

```diff
@@ -0,0 +1,22 @@
+package ar.edu.undec.adapter.service.curso.config;
+
+import curso.output.BuscarCursoPorIdRepository;
+import curso.output.CrearCursoRepository;
+import curso.usecase.BuscarCursoPorIdUseCase;
+import curso.usecase.crearcursousecase.CrearCursoUseCase;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+public class CursoBeanConfig {
+
+    @Bean
+    public CrearCursoUseCase crearCursoUseCase(CrearCursoRepository crearCursoRepository, BuscarCursoPorIdRepository buscarCursoPorIdRepository) {
```

Comment:
Idealmente, y para conseguir OpenClose, retornar la interfaz, no su implementación
```suggestion
    public CrearCursoInput crearCursoUseCase(CrearCursoRepository crearCursoRepository, BuscarCursoPorIdRepository buscarCursoPorIdRepository) {
```
Aplica igual para el Bean de BuscarCursoPorID